alerts/cluster-network-operator/NoRunningOvnMaster.md

```diff
@@ -0,0 +1,75 @@
+# NoRunningOvnMaster
+
+## Meaning
+
+This [alert][NoRunningOvnMaster] is triggered when there are no
+OVN-Kubernetes master control plane pods
+[Running][PodRunning]
+This is a critical-level alert if no OVN-Kubernetes master control plane pods
+are not running for `10m`.
+
+## Impact
+Networking control plane is not functional. Networking configuration updates
+will not be applied to the cluster.
+Without a functional networking control plane, existing workloads may continue
+to be partially functional,
+but new workloads will not be functional.
+Updates required for functioning Kubernetes services will not be performed.
+
+## Diagnosis
+### Cluster network operator
+Cluster Network operator (CNO) manages the CRUD operations for OVN-Kubernetes
+daemonset.
+Verify the CNO is running:
+
+    oc -n openshift-network-operator get pods -l name=network-operator
+
+Verify the [CNO](https://github.com/openshift/cluster-network-operator/) is
+functioning without error by checking the operator Status.
+You can check the Status of the CNO directly using the following command:
+
+    oc describe co network
+
+Pay attention to any error message reported by Status Conditions of type
+Degraded.
+
+Check the CNO logs for when it is reconciling the daemonset for ovnkube-master.
+
+    PODNAME=$(oc -n openshift-network-operator get pods
+                -l name=network-operator -o name)
+    oc -n  openshift-network-operator logs $PODNAME
+
+A successful reconcile for this daemonset looks like this in the CNO logs:
+
+    I0228 14:48:30.941130       1 log.go:184] reconciling (apps/v1, Kind=DaemonSet)
+                                openshift-ovn-kubernetes/ovnkube-master
+    I0228 14:48:30.960944       1 log.go:184] update was successful
+
+Verify the _DESIRED_ number of daemonsets is equal to the number of Kubernetes
+control plane nodes:
+
+    oc get nodes -l node-role.kubernetes.io/master="" -o name | wc -l
+    oc get ds -n openshift-ovn-kubernetes ovnkube-master
+
+### OVN-Kubernetes master pod
+Understand which container is failing in the OVN-Kubernetes master pod by
+describing one of the failing pods with `oc describe pod ...`.
+After understanding which container is not starting successfully, gather the
+runtime logs from that container.
+You may need to use `--previous` command with `oc get logs` command to get the
+logs of the previous execution run of a container.
+
+Pay close attention to any log output starting with "E" for Error:
+
+    oc -n openshift-ovn-kubernetes logs $OVNKUBE-MASTER-POD-NAME
+    --all-containers=true | grep "^E"
+
+## Mitigation
+
+The appropriate mitigation will be very different depending on the cause of the
+error discovered in the diagnosis.
+Investigate the issue using the steps outlined in diagnosis and contact the
+incident response team in your organisation if fixing the issue is not apparent.
```

Comment:
Would it be possible to cover some possible issue scenarios and solution(s) for it? Having references to some of the acceptable and safe solutions will be really helpful when the incident response team itself is changing from time to time. Covering most typical scenarios might suffice and we can add more later on when we come across it.

Reply:
I have never seen all the pods fail so I am unsure which scenario may cause it besides guessing things like internet connectivity pulling image or API server connectivity but they are scenarios that should not be added here.
I think we can revisit this when it actually occurs and add it later.

Reply:
Ack.. Main intent was to have clear actionable steps for any alert.. We can revisit this too in future..